Development

giulio-giunta opened the pull request:
Hi guys,
 
I added the manifests to enable logging and monitoring of the EKS cluster, and to create a storage class.
Also, I created two workfolders in Terraform Cloud associated with the "prod" and "dev" environments. The "prod" workfolder is linked to the master branch of this repository, so once this pull request is accepted it should kick a run in Terraform cloud. 
Finally I edited the READ.me. Please take a look.

Gulio

giulio-giunta committed:
added manifests for logging and monitoring

giulio-giunta committed:
changed instance type

giulio-giunta committed:
changed remote backend name

giulio-giunta committed:
removed remote backend

giulio-giunta committed:
changed cluster and vpc names, added storage class

giulio-giunta committed:
edited READ.me and changed folder name

jamesgeddes commented on .terraform.lock.hcl:
```diff
@@ -3,7 +3,7 @@
 provider "registry.terraform.io/hashicorp/aws" {
   version     = "3.23.0"
-  constraints = ">= 3.3.0, >= 3.20.0"
+  constraints = ">= 3.20.0, >= 3.22.0"
```
Can we use the [pessimistic constraint operator](https://www.terraform.io/docs/language/expressions/version-constraints.html#gt--1) please? This will allow us to check major changes before implementing them.

giulio-giunta replied:
The constraints for all providers are specified in the versions.tf file. 
The AWS constraint cannot be set to "=" otherwise when you run terraform apply you'll get an error, because some components of the cluster require different versions (I tried it).

jamesgeddes commented on eks-cluster.tf:
```diff
@@ -17,17 +16,19 @@ module "eks" {
-      instance_type                 = "t3.small"
+      instance_type                 = "t3.medium"
       additional_userdata           = "echo foo bar"
       asg_desired_capacity          = 2
```
Can we put these in different availability zones?

giulio-giunta replied:
availability_zones            = "data.aws_availability_zones.available.names"
I added the above line in the eks-cluster.tf file: that should do the job.

jamesgeddes commented on kubeconfig_education-eks-YBEeiURS:
```diff
@@ -0,0 +1,28 @@
+
+contexts:
+- context:
+    cluster: eks_education-eks-YBEeiURS
```
is this a sandbox instance or a working instance?

giulio-giunta replied:
It's a sandbox instance until the pull request is not approved and merged to master. Once that happens becomes a production instance. Before merging to master the workspace indicated in the remote backend block in the eks-cluster.tf file need to be changed from "dev" to "prod"

jamesgeddes commented on kubeconfig_education-eks-YBEeiURS:
```diff
@@ -0,0 +1,28 @@
+clusters:
+- cluster:
+    server: https://D4A69E621E273C7ACCE3E12003FEF15D.gr7.eu-west-2.eks.amazonaws.com
+    certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUN5RENDQWJDZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRJeE1ERXpNVEV6TXpFd01Wb1hEVE14TURFeU9URXpNekV3TVZvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTWhCCkRQSW5icjR2bXBpVlJvOEo5WnJpOGNQUnJjZDdBZENmU3FJUmplRFdYbzF4RTZGcDRxcFA ...
```
A silly question just to be clear - this is not sensitive, is it?

giulio-giunta replied:
It is, it's just encoded with base64 tool

giulio-giunta committed:
added Carwyn in the aws-auth configMap and configuration to spread to worker nodes in different AZs

giulio-giunta committed:
added Carwyn in the aws-auth configMap and configuration to spread to worker nodes in different AZs

giulio-giunta committed:
replaced asg_desired_capacity with  asg_min_size and asg_max_size, removed random provider

giulio-giunta commented:
Hi guys, 

I was reading in the Terraform github repo and I found out that Terraform does not support changing the asg_desired_capacity. Here is the document: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md 
Also,  ASG is not configured to be recreated when the launch configuration or template changes. For this reason the tags applied to the ec2 instances within the worker_group don't get updated, whereas the tag applied to the eks cluster does get updated. I attached a couple of pictures to show that. 
![Screenshot_2021-02-17_21-41-54](https://user-images.githubusercontent.com/56529931/108271642-02f44600-7169-11eb-997b-a2fc160c178b.png)
![Screenshot_2021-02-17_21-41-31](https://user-images.githubusercontent.com/56529931/108271656-0982bd80-7169-11eb-8d2e-2b0709f9980e.png)

jamesgeddes commented:
Wow, that is a serious limitation of TF then. Did you log a ticket to encourage Hashicorp to fix this?

giulio-giunta commented:
Someone alread logged a ticket. The reason they do it is because changing
the asg could break the Kubernetes cluster.

On Wed, 17 Feb 2021, 21:51 James Geddes, <notifications@github.com> wrote:

> Wow, that is a serious limitation of TF then. Did you log a ticket to
> encourage Hashicorp to fix this?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GeekZoneHQ/infra/pull/4#issuecomment-780878087>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANPJIC3UBS5FLVE42DDGIWDS7Q26RANCNFSM4W6LESWA>
> .
>

jamesgeddes commented:
Ah super. Can you link the ticket here so that we can show it is needed?

giulio-giunta commented:
Hi James,

here is the link to the issue with asg that has already been raised:
https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1223


On Thu, 18 Feb 2021 at 04:57, James Geddes <notifications@github.com> wrote:

> Ah super. Can you link the ticket here so that we can show it is needed?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GeekZoneHQ/infra/pull/4#issuecomment-781053805>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANPJIC6SIR5YXITRJQQ77BDS7SM4NANCNFSM4W6LESWA>
> .
>

giulio-giunta committed:
added kubernetes resource definitions for horizontal pod autoscaling and metric-server

giulio-giunta committed:
added ingress controller with tls-termination and ingress resources to setup SSL for test.geek.zone, logging.geek.zone and monitoring.geek.zone

giulio-giunta committed:
added terraform configuration files for integration between EKS cluster and RDS Postgres database. Also added ingress configuration files to have a unique load balancer routing external traffic to various services inside the kubernetes cluster. Configured the django webapp deployment with environment variables to access the database

giulio-giunta committed:
arranged the Kubernetes resource definition files into separate folders, each referring to a different namespace in the cluster; added liveliness and readiness probes and the reference to secrets in the webapp deployment; added configuration for secrets envelope encryption in ETCD datastore; rearranged variables in Terraform and gave access to the EKS cluster to another user

giulio-giunta commented:
Hi guys, 
Yesterday I rearranged the Kubernetes resource definition files into separate folders, each referring to a different namespace in the cluster; added liveliness and readiness probes and the reference to secrets in the webapp deployment; added configuration for secrets envelope encryption in ETCD datastore; rearranged variables in Terraform and gave access to the EKS cluster to another user

giulio-giunta committed:
This adds the following configuration: creates a celery namespace where celery workers run indipendently from the django backend; it adds configuration for Keda pod autoscaler based on requests to rabbitmq; it updates the metrics-server CRDS and deleted the folder containing the old metrics-server files; it adds secrets entries in .gitignore to avoid sending secrets to remote repository

giulio-giunta committed:
arranged the Kubernetes resource definition files into separate folders, each referring to a different namespace in the cluster; added liveliness and readiness probes and the reference to secrets in the webapp deployment; added configuration for secrets envelope encryption in ETCD datastore; rearranged variables in Terraform and gave access to the EKS cluster to another user

giulio-giunta committed:
This adds the following configuration: creates a celery namespace where celery workers run indipendently from the django backend; it adds configuration for Keda pod autoscaler based on requests to rabbitmq; it updates the metrics-server CRDS and deleted the folder containing the old metrics-server files; it adds secrets entries in .gitignore to avoid sending secrets to remote repository.

giulio-giunta committed:
This adds basic authentication to logging.geek.zone; it changes image tag in the deployments in both celery and web namespaces;  finally it fixes some errors occurred when running terraformplan due to Terraform RDS modules update

giulio-giunta committed:
This adds circleci configuration and slightly modifies README.md file

giulio-giunta committed:
Add .circleci/config.yml

giulio-giunta committed:
This changes circleci configuration

giulio-giunta committed:
Merge branch 'development' of https://github.com/GeekZoneHQ/infra into development

giulio-giunta committed:
This changes Terraform orb identifier

giulio-giunta committed:
This changes Terraform version in circleci config.yml

jamesgeddes commented:
@giulio-giunta is this still ongoing?

giulio-giunta committed:
This adds k8s manifests that match the current workloads running in the cluster and some options to run a Valheim server in k8s

giulio-giunta committed:
Merge branch 'development' of https://github.com/GeekZoneHQ/infra into development

giulio-giunta committed:
This removes unnecessary ci/cd config

giulio-giunta committed:
This adds configuration to launch a Geek.Zone cluster in azure aks

giulio-giunta committed:
This changes the folder layout and adds scripts to create and destroy infrastructure in aws

giulio-giunta committed:
This removes Valheim helm chart and k8s folders from the project

giulio-giunta committed:
This adds scripts in azure folders and fixes scripts in aws folder

giulio-giunta committed:
This updates the README.md

giulio-giunta committed:
This fixes README.md

giulio-giunta committed:
This fixes scripts

giulio-giunta committed:
This removes sleep after terraform apply

giulio-giunta committed:
This updates scripts in aws and azure folders

giulio-giunta committed:
This adds configuration for ingress controller and tls encryption

giulio-giunta committed:
This adds secrets in gitignore

giulio-giunta committed:
This adds wildcard certificate in k8s cluster

giulio-giunta committed:
This removes folder that does not update in remote repo

giulio-giunta committed:
This restores deleted folder

giulio-giunta committed:
This changes k8s version to fix issue with external-dns, adjusts resource limits and improves script to launch a cluster

giulio-giunta committed:
This changes k8s version to fix issue with external-dns, adjusts resource limits and improves script to launch a cluster

giulio-giunta committed:
This adds external-dns resources in script
